### PR TITLE
docs(task-forms): document dynamic forms

### PR DESCRIPTION
**Pull request by tasso94**
related to CAM-14887

### CODE REVIEW

File: content/user-guide/task-forms/_index.md [hunk 1]
```diff
@@ -154,6 +154,44 @@ The submitted values of a form are returned as variables to the process engine:
...
 
+## Dynamic Components
+
+You can bind the available options of some component types (Select, Radio Buttons, Checklist, and Taglist) to a variable.
+Like this, Camunda Forms show available options dynamically based on process data (variables).
+
+If you want to bind a variable to a dynamic component, define its name in the `valuesKey` field 
+of Camunda Modelers form builder for the respective component.
+
+Camunda Forms support the following variable types that can represent JSON:
```
Comment (danielkelemen) [inline content/user-guide/task-forms/_index.md:165]: I know Forms is plural but "Camunda Forms" is one tool/feature. Shouldn't we use third-person singular? I'm not sure.

```suggestion
Camunda Forms supports the following variable types that can represent JSON:
```

**Review by danielkelemen — APPROVED** [review]
I left one minor note but otherwise looks good to me!

File: content/user-guide/task-forms/_index.md (hunk 1 above)
Comment (tasso94) [reply content/user-guide/task-forms/_index.md:165]: True!

**Comment by christinaausley** [comment]
I'll retroactively give this a quick review and PR technical changes if necessary 👍

**Pull request merged**